### PR TITLE
Document how mods can integrate with the additional features of this NEI fork

### PR DESCRIPTION
**Pull request by wlhlm**
This has been spurred by #394, #386, and #385.

Basically, our NEI fork adds a bunch of features that require specifying additional metadata for recipe handlers such as an icon and width and height. This can be done via CSV, IMC, or event handling, but none of it is documented. Even looking at the source code, it can be quite the effort for someone externally to figure out how our new features are supposed to be supported in third party mods. This PR tries to improve this situation by adding metadata specs and coding guidelines to the README and javadocs to various APIs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,12 +24,192 @@ If you have issues with NEI outside of the GTNH modpack you may report them in t
...
+
+#### I cannot update my mod, because it is no longer maintained, and I'm not a developer
+
+Some mods are no longer maintained and cannot be updated due to licensing. For those we provide a last-ditch effort called the "height hack". This tries to emulate the old NEI recipe layout, specifically on the vertical axis where our version of NEI had the most significant changes. This should help with misaligned tooltips and click zones especially.
+
+The height hack can be enabled by adding the fully-qualified class name of the recipe handler to the `heighthackhandlers.cfg` config file. If you encounter a recipe handler that requires this measure please open an issue and let us know.
+
+#### I'm a developer working on my own mod or maintaining a forked mod
+
+Please consider adding support for our IMC messages or events to automatically register handler metadata for your mod. Alternatively, we can provide default metadata for your recipe handlers in `assets/nei/csv/handlers.csv`, let us know the parameters, or even better, open a pull request. The fields available in the new metadata are described in the following section.
```
Comment (mitchej123) [inline README.md:48]: I prefer _not_ allowing handlers.csv for actively maintained mods, they should use IMC or the API.  handlers.csv is for unmaintained/legacy mods.

**Review by mitchej123 — APPROVED** [review]
Looks good, thanks for putting this together.  Would prefer updating the active mod to _not_ entertain the csv files, other than that it's good.

**Comment by wlhlm** [comment]
@mitchej123 Thanks for the feedback. There still more stuff I'm working on for this PR, especially around the `handler` and `handlerID` columns in `catalysts.csv` and `handlers.csv`. I'm considering renaming either one of them or both to reduce confusion. As well as even more documentation.

File: src/main/java/codechicken/nei/config/IMCHandler.java [hunk 2]
```diff
@@ -126,32 +145,56 @@ private static void handleRegisterCatalystInfo(IMCMessage message) {
...
             processedCatalystSenders.add(message.getSender());
         }
         final NBTTagCompound tag = message.getNBTValue();
-        final String handlerID = tag.getString("handlerID");
-        if (handlerID.isEmpty()) {
-            NEIClientConfig.logger.warn("Missing handlerID for registerCatalystInfo!");
+        final String catalystHandlerID;
+        if (tag.hasKey("handlerID")) {
+            NEIClientConfig.logger.warn(
+                    "Received tag 'handlerID' for registerCatalystInfo IMC from {}, this is deprecated, please use 'catalystHandlerID' instead",
```
Comment (Yoghurt4C) [inline src/main/java/codechicken/nei/config/IMCHandler.java:154]: revert this frivolous change

Comment (wlhlm) [reply src/main/java/codechicken/nei/config/IMCHandler.java:154]: I have been trying to reduce potential confusion from using our APIs. Unfortunately, `handlerID` here can be confused with the old `handler` field in the register handler info IMC, despite the former allowing for a wider range of potential values (I've documented the details in the README), that's why I have decided to rename it.

BTW, I'm working on updating the Et Futurum GTNH NEI handlers to fix some of the issues with the integration and to use as a case study.

Comment (wlhlm) [reply src/main/java/codechicken/nei/config/IMCHandler.java:154]: I should make the log message clearer that it's just a renaming.

Comment (Yoghurt4C) [reply src/main/java/codechicken/nei/config/IMCHandler.java:154]: in this very moment, you are Ϡ

Comment (connor135246) [reply src/main/java/codechicken/nei/config/IMCHandler.java:154]: The message says it's "deprecated", does that mean that the old name will stop working in a future version? That seems like a bad idea for maintaining compatibility with mods that may not be updated in the future.
And if the old name won't stop working in a future version, is there a reason to log a warning at all?

**Comment by Roadhog360** [comment]
Is it possible to get a list of catalysts from a handler? I've been wondering how to do this but I've never been able to get an answer.

**Comment by boubou19** [comment]
@wlhlm any news on this? or should we close it?

**Comment by serenibyss** [comment]
Closing for now due to inactivity and merge conflicts, feel free to reopen

**Comment by Roadhog360** [comment]
In what way is #385 misunderstanding the docs? If anyone could explain I'd appreciate it. As far as I can tell, the stuff that I referenced in that issue still functions perfectly correctly and as intended. The catalyst handler is loaded and works just fine, yet the logs claim it failed even though it didn't.

**Pull request closed without merging**